IMPORTED_LOCATION not set for imported target "Upstream::amici"

Since [today](https://github.com/AMICI-dev/AMICI/actions/runs/9073921357/job/24931811299) on GHA with cmake==3.29.3:

```
CMake Error in CMakeLists.txt:
  IMPORTED_LOCATION not set for imported target "Upstream::amici"
  configuration "Debug".


CMake Error in CMakeLists.txt:
  IMPORTED_LOCATION not set for imported target "Upstream::amici"
  configuration "Debug".


-- Generating done (0.0s)
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/scripts/installAmiciSource.sh
+++ b/scripts/installAmiciSource.sh
@@ -33,7 +33,7 @@ fi
 export PYTHON_EXECUTABLE="${AMICI_PATH}/venv/bin/python"
 
 python -m pip install --upgrade pip wheel
-python -m pip install --upgrade pip setuptools cmake_build_extension numpy
+python -m pip install --upgrade pip setuptools cmake_build_extension==0.5.1 numpy
 python -m pip install git+https://github.com/FFroehlich/pysb@fix_pattern_matching # pin to PR for SPM with compartments
 AMICI_BUILD_TEMP="${AMICI_PATH}/python/sdist/build/temp" \
   python -m pip install --verbose -e "${AMICI_PATH}/python/sdist[petab,test,vis]" --no-build-isolation

